ibm_driver.c

```diff
@@ -142,7 +146,9 @@ static int dbh_prepare_stmt(pdo_dbh_t *dbh, pdo_stmt_t *stmt, const char *stmt_s
 	SQLSMALLINT server_len = 0;
 
 	/* in case we need to convert the statement for positional syntax */
+#if !(PHP_MAJOR_VERSION > 8 || (PHP_MAJOR_VERSION == 8 && PHP_MINOR_VERSION == 1))
```

Comment:
I didn't realize there was a `!` at the beginning at first. Since there's gonna be lot of these checks, might be worth adding a definition:
```c
#if PHP_MAJOR_VERSION > 8 || (PHP_MAJOR_VERSION == 8 && PHP_MINOR_VERSION == 1)
#define PHP_8_1_OR_HIGHER
#endif
```

Then you can just use `#ifdef PHP_8_1_OR_HIGHER` and `#ifndef PHP_8_1_OR_HIGHER` to make things more clear.

Reply:
it should be

```
#if PHP_MAJOR_VERSION > 8 || (PHP_MAJOR_VERSION == 8 && PHP_MINOR_VERSION >= 1)
#define PHP_8_1_OR_HIGHER
#endif
```